src/Orleans/Configuration/ClusterConfiguration.cs

```diff
@@ -441,7 +441,7 @@ internal static IPAddress ResolveIPAddress(string addrOrHost, byte[] subnet, Add
                 }
 
                 var candidates = new List<IPAddress>();
-                IPAddress[] nodeIps = Dns.GetHostAddresses(addrOrHost);
+                IPAddress[] nodeIps = Dns.GetHostAddressesAsync(addrOrHost).GetAwaiter().GetResult();
```

Comment:
Very good find @dVakulen! As for the more general issue of *async*, I'd advocate doing it even only for the sake of being asynchronous. I know this might be an overly broad statement, but I have actually one particular piece of code in mind at [IGatewayListProvider](https://github.com/dotnet/orleans/blob/fc0af9079413505de6c5dad928a266fccda74d68/src/Orleans/Messaging/IGatewayListProvider.cs#L47) and the implementation in relation storage [SqlMembershipTable.GetGateways()](https://github.com/dotnet/orleans/blob/538e579e926ca7a12c310be2e3fa978e95a4ee6c/src/OrleansSQLUtils/Messaging/SqlMembershipTable.cs#L104). I might as well as fix this as per this comment later today and leave a lingering point it'd probably in order to change this function to an asynchronous one.